Marlin/src/gcode/feature/trinamic/M919.cpp

```diff
@@ -0,0 +1,257 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2021 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../../../inc/MarlinConfig.h"
+
+#if HAS_TRINAMIC_CONFIG
+
+#include "../../gcode.h"
+#include "../../../feature/tmc_util.h"
+#include "../../../module/stepper/indirection.h"
+
+#define DEBUG_OUT ENABLED(MARLIN_DEV_MODE)
+#include "../../../core/debug_out.h"
+
+/**
+ * M919: Set TMC stepper drive chopper times
+ *
+ * Parameters:
+ *   XYZIJKE...
+ *   I[index]    - Axis sub-index (Omit or 0 for X, Y, Z, E0; 1 for X2, Y2, Z2, E1; 2 for Z3, E2; 3 for Z4, E3; 4 for E4; 5 for E5; 6 for E6; 7 for E7)
+ *   T[index]    - Extruder index (Zero-based. Omit for E0 only.)
+ *   O           - time-off         [ 1..15]
+ *   V           - hysteresis_end   [-3..12]
+ *   S           - hysteresis_start [ 1...8]
+ * 
+ * With no parameters report chopper times for all axis
+ */
+
+void GcodeSuite::M919() {
+  #define TMC_SAY_CHOPPER_TIME(Q) tmc_print_chopper_time(stepper##Q)
+  #define TMC_SET_CHOPPER_TIME(Q) stepper##Q.set_chopper_times(time_off, hysteresis_end, hysteresis_start)
```

Comment:
I used same approach as M906

Reply:
No worries. Use of convenience macros in this way is encouraged, and such macros are even a necessity in some places so that the code can adapt to extra linear axes (now XYZ plus 3 and soon XYZ plus 6) via the `LINEAR_AXIS_*` and `REPEAT` macros.

The extra benefits here are also for your carpal tunnels. If the signature of `set_chopper_times` were to ever change, we need only edit one line, whereas if every instance were replaced with a function call, every instance would need to be edited.